### PR TITLE
Store rollback

### PR DESCRIPTION
**Pull request by stroem**

### CODE REVIEW

**Comment by stroem** [comment]
Tests are not fixed yet.

File: eventsource/repository.go [hunk 1]
```diff
@@ -18,12 +18,17 @@ var (
...
-	Save(records ...Record) error
-	SaveWithContext(ctx context.Context, records ...Record) error
-	Load(id string) (record []Record, err error)
-	LoadWithContext(ctx context.Context, id string) (record []Record, err error)
+	NewTransaction(ctx context.Context, records ...Record) (StoreTransaction, error)
+	Load(ctx context.Context, id string) (record []Record, err error)
 }
 
+type StoreTransaction interface {
+	Save() error
```
Comment (lonnblad) [inline eventsource/repository.go]: `Commit`?

File: eventsource/repository.go [hunk 2]
```diff
@@ -104,10 +110,17 @@ func (repo *repository) SaveWithContext(ctx context.Context, events ...Event) (e
...
+		return
+	}
+
+	if err = tx.Save(); err != nil {
 		return
 	}
 
+	rollback = tx.Rollback
+
 	for idx, record := range records {
```
Comment (lonnblad) [inline eventsource/repository.go]: it looks like `OnSave` will be called even if a rollback is executed, would that be a problem?

Comment (stroem) [reply eventsource/repository.go]: Yeah, that could be a problem. I solved it, no need to keep it. Most of the time you want to pass other arguments to OnSave, therefore you can do this outside the package, before calling Save or after.

**Pull request merged**